xarray/backends/common.py

```diff
@@ -503,3 +503,43 @@ def assert_open(self):
         if not self._isopen:
             raise AssertionError('internal failure: file must be open '
                                  'if `autoclose=True` is used.')
+
+
+class PickleByReconstructionWrapper(object):
+
+    def __init__(self, opener, *args, **kwargs):
+
+        self.opener = opener
+        self.open_args = args
+        self.open_kwargs = kwargs
```

Comment:
For simplicity, let's pool these into a single `functools.partial` object. Note that you can get out keyword arguments if desired from `partial.keywords`.

Reply:
should we just have the backends pass in a partial function then?

Reply:
If we don't anticipate adding any argument, then I think it's just as easy to make the partial function here. I suppose there's some potential overlap in needs with auto-closing a file, so I'm OK either way.